bindings/python/FloatingBaseEstimators/src/FloatingBaseEstimators.cpp

```diff
@@ -25,8 +27,126 @@ void CreateKinDynComputations(pybind11::module& module)
+             })
+        .def("get_center_of_mass_position",
+             [](KinDynComputations& impl) {
+                 Eigen::VectorXd pos(3);
+                 if (!impl.getCenterOfMassPosition(pos))
+                 {
+                     throw py::value_error("Failed to get the center of mass position.");
+                 }
+                 return pos;
+             })
+        .def("get_center_of_mass_velocity",
+             [](KinDynComputations& impl) {
+                 Eigen::VectorXd vel(3);
+                 if (!impl.getCenterOfMassVelocity(vel))
+                 {
+                     throw py::value_error("Failed to get the center of mass velocity.");
+                 }
+                 return vel;
+             })
+        .def("get_world_transform",
+             [](KinDynComputations& impl, std::string name) {
+                 Eigen::Matrix<double, 4, 4> world_T_frame;
+                 if (!impl.getWorldTransform(name, world_T_frame))
+                 {
+                     throw py::value_error("Failed to get the trasform for the specified frame.");
+                 }
+                 return world_T_frame;
+             })
+        .def("get_world_base_transform",
+             [](KinDynComputations& impl) {
+                 Eigen::Matrix<double, 4, 4> world_T_base;
+                 if (!impl.getWorldBaseTransform(world_T_base))
+                 {
+                     throw py::value_error("Failed to get the trasform for the base.");
+                 }
+                 return world_T_base;
+             })
+        .def(
+            "set_joint_pos",
+            [](KinDynComputations& impl, Eigen::Ref<const Eigen::VectorXd> s) {
+                return impl.setJointPos(iDynTree::make_span(s.data(), s.size()));
+            },
+            py::arg("s"))
+        .def(
+            "set_floating_base",
+            [](KinDynComputations& impl, const std::string & s) {
+                return impl.setFloatingBase(s);
+            })
+        .def("get_robot_state",
+             [](KinDynComputations& impl) {
+                 Eigen::Matrix<double, 4, 4> world_T_base;
+                 Eigen::VectorXd s(impl.getNrOfDegreesOfFreedom());
+                 Eigen::VectorXd base_velocity(6);
+                 Eigen::VectorXd s_dot(impl.getNrOfDegreesOfFreedom());
+                 Eigen::VectorXd world_gravity(3);
+                 if (!impl.getRobotState(world_T_base, s, base_velocity, s_dot, world_gravity))
+                 {
+                     throw py::value_error("Failed to get the robot state.");
+                 }
+                 KinDynKinematicsRobotState robot_state;
+                 robot_state.baseTransform = world_T_base;
+                 robot_state.jointPositions = s;
+                 robot_state.baseVelocity = base_velocity;
+                 robot_state.jointVelocities = s_dot;
+                 robot_state.worldGravity = world_gravity;
+                 return robot_state;
```

Comment:
Nit: you can maybe save the creation of tmp objects:

```cpp
KinDynKinematicsRobotState robot_state(impl.getNrOfDegreesOfFreedom());

if (!impl.getRobotState(robot_state.baseTransform, robot_state.jointPositions, robot_state.baseVelocity, robot_state.jointVelocities, robot_state.worldGravity))
{
    throw py::value_error("Failed to get the robot state.");
}

```

Reply:
You're right, done via 7e57363665a649ad7ba97f209189c724af0f1165.